dynamiqs/time_array.py

```diff
@@ -376,12 +379,16 @@ def conj(self) -> _Factor:
         return _PWCFactor(self.times, self.values.conj())
 
     def __call__(self, t: Scalar) -> Array:
-        if t < self.times[0] or t >= self.times[-1]:
-            return jnp.zeros_like(self.values[..., 0])  # (...)
-        else:
-            # find the index $k$ such that $t \in [t_k, t_{k+1})$
-            idx = jnp.searchsorted(self.times, t, side='right') - 1
-            return self.values[..., idx]  # (...)
+        idx = jnp.searchsorted(self.times, t, side='right') - 1
+        return jax.lax.select(
+            t < self.times[0],
+            jnp.zeros_like(self.values[..., 0]),
+            jax.lax.select(
+                t >= self.times[-1],
+                jnp.zeros_like(self.values[..., 0]),
+                self.values[..., idx],
+            ),
+        )
```

Comment:
I think https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.cond.html#jax.lax.cond is better suited here.
> In general [select()](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.select.html#jax.lax.select) leads to evaluation of both branches, although the compiler may elide computations if possible. For a similar function that usually evaluates only a single branch, see [cond()](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.cond.html#jax.lax.cond).

Reply:
Also there's no need to nest select, and `jnp.searchsorted` should only be called in the second branch (as in the original Python code).

Reply:
Adressed in https://github.com/dynamiqs/dynamiqs/pull/411